### PR TITLE
Doc Makefile clean command refactored and softmax function detailed documentation

### PR DESCRIPTION
**Pull request by marquesVF**
- I've made some changes in the _doc/Makefile_ document in order for it to make a proper cleaning of generated files.

- The _softmax_ function was lacking of documentation.

Should _softmax_ function documentation be exposed at the online documentation web site?

### CODE REVIEW

**Comment by the872** [comment]
Just curious, why isn't softmax inside the GLM class?

File: doc/Makefile [hunk 1]
```diff
@@ -40,9 +40,10 @@ help:
 .PHONY: clean
 
 clean:
-	rm -rf $(BUILDDIR)/*
+	rm -rf $(BUILDDIR)/
```
Comment (jasmainak) [inline doc/Makefile]: so you delete the folder too? why is that?

File: pyglmnet/pyglmnet.py [hunk 2]
```diff
@@ -38,8 +38,17 @@ def set_log_level(verbose):
...
+    """Softmax function of given array of number w
+
+    Parameters
+    ----------
+    w: array
+        any object exposing the array interface
+
+    Returns
+    -------
+    dist: ndarray
```
Comment (jasmainak) [inline pyglmnet/pyglmnet.py]: let's stick to `array` everywhere?

File: doc/Makefile (hunk 1 above)
Comment (marquesVF) [inline doc/Makefile]: I was thinking about delete any file or folder created by the build process.

Comment (jasmainak) [inline doc/Makefile]: okay, as long as you don't break `make html`, fine with me. Did you try running `make html` after `make clean` to see if it still works?

**Comment by jasmainak** [comment]
> Should softmax function documentation be exposed at the online documentation web site?

@pavanramkumar is this meant to be part of the public API? I suppose we should make it private. I don't see how the users will be using it ...

> General question: why isn't softmax inside the GLM class?

I suppose `softmax` could be used by another class (if there was one) and there is nothing specific to GLM about it. That's why it's a separate function ...

**Comment by pavanramkumar** [comment]
@jasmainak @marquesVF @the872 agreed that softmax function should be outside the GLM class and potentially private. it's not necessary for the user. i'd say same for label binarizer.

i'm in the process of moving helper functions to a separate file called `utils.py`. 

see here: https://github.com/pavanramkumar/pyglmnet/blob/0a1f7f8d52897e37eb408428f3bf15ac58b24f6a/pyglmnet/pyglmnet_utils.py

File: doc/Makefile (hunk 1 above)
Comment (marquesVF) [inline doc/Makefile]: Yes. I tested the build process after I've changed the `make clean` and it worked normally.

File: pyglmnet/pyglmnet.py (hunk 2 above)
Comment (marquesVF) [inline pyglmnet/pyglmnet.py]: I'll refactor it.

**Comment by jasmainak** [comment]
okay cool, I merged this branch after cosmit https://github.com/pavanramkumar/pyglmnet/commit/516d5227ea5427addef40e6d898854587519cb5a. Thanks @marquesVF

**Pull request merged**